cypress/integration/RichTextEditor.spec.ts

```diff
@@ -808,4 +808,273 @@ describe('Rich Text Editor', () => {
       ].map((type) => getDropdownItem(type).get('button').should('not.be.disabled'));
     });
   });
+
+  describe('Links', () => {
+    const getLinkTextInput = () => cy.findByTestId('link-text-input');
+    const getLinkTypeSelect = () => cy.findByTestId('link-type-input');
+    const getLinkTargetInput = () => cy.findByTestId('link-target-input');
+    const getSubmitButton = () => cy.findByTestId('confirm-cta');
+    const getEntityTextLink = () => cy.findByTestId('cf-ui-form').findByTestId('cf-ui-text-link');
+    const expectDocumentStructure = (...nodes) => {
+      expectRichTextFieldValue(
+        doc(
+          block(
+            BLOCKS.PARAGRAPH,
+            {},
+            ...nodes.map(([nodeType, ...content]) => {
+              if (nodeType === 'text') return text(...content);
+              const [data, textContent] = content;
+              return inline(nodeType, data, text(textContent));
```

Comment:
I've updated this test to account for the new document structure in which text nodes are direct descendants of hyperlinks (rather than having a `hyperlink` -> `paragraph` -> `text` hierarchy).